geometry/src/main/java/coppercore/Line.java

```diff
@@ -0,0 +1,31 @@
+package coppercore.geometry;
+
+public class Line {
+    private double x1;
+    private double y1;
+    private double x2;
+    private double y2;
+    private double slope;
+
+    public Line(double x1, double y1, double x2, double y2) {
+        this.x1 = x1;
+        this.y1 = y1;
+        this.x2 = x2;
+        this.y2 = y2;
+        this.slope = (y2 - y1) / (x2 - x1);
```

Comment:
What if x2==x1?

(The answer is probably to use linear interpolation and parameterize the line rather than using slope intercept form)

Reply:
how would i go about using linear interpolation?

Reply:
So I found a better answer actually:

For 3 points (A, B, and C) if you have distance(A,B) = distance (A, C) + distance(C, B), then C is on the line between A and B

So you just need to do the distance formula a few times, which honestly should be a function of a `Line` anyway.